### PR TITLE
Deduplicate test_basic arrayimagecopy

### PR DESCRIPTION
**Pull request by JohnKesapidesARM**
Merge test_arrayimagecopy3d and test_arrayimagecopy 
as they share a lot of common code.

Signed-off-by: John Kesapides <john.kesapides@arm.com>

### CODE REVIEW

File: test_conformance/basic/test_arrayimagecopy.cpp [hunk 1]
```diff
@@ -14,134 +14,173 @@
...
-            break;
+    cl_uchar *bufptr, *imgptr;
+    clMemWrapper buffer, image;
+    int img_width = 512;
+    int img_height = 512;
+    int img_depth = (image_type == CL_MEM_OBJECT_IMAGE3D) ? 32 : 1;
+    size_t elem_size;
+    size_t buffer_size;
+    cl_int err;
+    cl_event copyevent;
```
Comment (EwanC) [inline test_conformance/basic/test_arrayimagecopy.cpp:42]: Idea for a follow-on PR that we could use a `clEventWrapper` round this event, same for PR  https://github.com/KhronosGroup/OpenCL-CTS/pull/1538/

**Review by EwanC — APPROVED** [review]

File: test_conformance/basic/test_arrayimagecopy.cpp (hunk 1 above)
Comment (EwanC) [reply test_conformance/basic/test_arrayimagecopy.cpp:42]: Awesome, thanks ⭐

**Comment by lakshmih** [comment]
Qualcomm is okay with this change

Sreelakshmi Haridas Maruthur
Qualcomm

**Review by lakshmih — APPROVED** [review]

**Comment by bashbaug** [comment]
Has two reviews, merging.

**Pull request merged**